src/test/java/de/tum/cit/aet/artemis/localvcci/LocalVCLocalCIIntegrationTest.java

```diff
@@ -929,21 +951,32 @@ void testFetchPush_instructorPracticeRepository() throws Exception {
     @WithMockUser(username = TEST_PREFIX + "student1", roles = "USER")
     void testBuildPriorityAfterDueDate() throws Exception {
         // Set dueDate before now
+        sharedQueueProcessingService.removeListener();
         programmingExercise.setDueDate(ZonedDateTime.now().minusMinutes(1));
         programmingExerciseRepository.save(programmingExercise);
 
         testPriority(instructor1Login, PRIORITY_OPTIONAL_EXERCISE);
+
+        queuedJobs.clear();
+        sharedQueueProcessingService.init();
```

Comment:
It would be better to move the tests that need such a setup into a `@Nested` subclass and remove and re-add the listener in `@BeforeEach` and `@AfterEach` methods.

Reply:
That's a good idea, I'll do that